### ISSUE TITLE
About original CenterNet performance on DOTA

### ISSUE THREAD

**Issue by Tarazed** [yijingru/BBAVectors-Oriented-Object-Detection#12]
Hello, Thanks for your nice work again!
I also want to adopt CenterNet by just adding an extra angle prediction branch as which mentioned in your paper. For data, I used the same data as BBAVecotrs, except for negative images that do not contain any objects. I changed the backbone from hourglass104 into ResNet101 and upsampling layers as the same as BBAVectors. The losses are just followed the original CenterNet and the weight of angle loss is 0.1. After training for 120 epochs, I got the mAP of 0.62, which is lower than your result of 0.6887.
Did I miss some key steps? Or the negative images are very important to improve the performance?

**Comment by yijingru** [comment]
I didn't remove the negative images from the output of image split algorithms. For center+wh+theta,  I used the same training skills and architectures except for the output branches for a fair comparison.

**Comment by Fly-dream12** [comment]
I also wonder how to adopt CenterNet by adding an extra angle prediction branch, could you mention the code of this part?
Thanks @ yijingru @Tarazed

**Comment by yijingru** [comment]
Change `heads` in the `main.py`, and also change the ground-truth accordingly (`wh` is the width and height of the box, `angle` is `theta` in Fig1a.  I used `SmoothL1` loss for angle regression. And box decoding parts.
`    heads = {'hm': num_classes[args.dataset],
             'wh': 2,
             'reg': 2,
             'angle': 1
}`

**Comment by Tarazed** [comment]
> Change `heads` in the `main.py`:
> 
> ```
>     heads = {'hm': num_classes[args.dataset],
>              'wh': 2,
>              'reg': 2,
>              'angle': 1
>              }
> ```
> 
> and change the ground-truth accordingly (`wh` is the width and height of the box, `angle \in [-90,0)` is `theta` in Fig1a, note ground-truth `theta` is already calculated in `datasets/base.py`. I used `SmoothL1` loss for angle regression.

I have found a strange phenomenon: when I used the 3rd channel of 'wh' head to predict angle, I got 0.636 of mAP, but when I  used an extra branch to predict angle as you said, I got 0.530 mAP... I set the random seed to 0, the performance of which is a little bit lower than random seed. I'm confusing as extra branch should be better.

**Comment by yijingru** [comment]
Maybe training loss matters. I used weighted loss as`loss =  hm_loss + wh_loss + off_loss + 0.1*angle_loss`.

**Comment by Tarazed** [comment]
> Maybe training loss matters. I used weighted loss as`loss = hm_loss + wh_loss + off_loss + 0.1*angle_loss`. The multi-tasks are hard to optimize sometimes. But we do see the angle is hard to learn for the network.

That's awesome! The loss weight of multi-task can significantly affect the results! After changed the weight of theta_loss, I got 0.68mAP! Thank you very much!

**Comment by yijingru** [comment]
> > Maybe training loss matters. I used weighted loss as`loss = hm_loss + wh_loss + off_loss + 0.1*angle_loss`. The multi-tasks are hard to optimize sometimes. But we do see the angle is hard to learn for the network.
> 
> That's awesome! The loss weight of multi-task can significantly affect the results! After changed the weight of angle_loss, I got 0.68mAP! Thank you very much!

My pleasure.

**Comment by yingyu13** [comment]
i also want to know the loss weight of multi-task of 0.68mAP, can you share it? thanks very much! @Tarazed

**Comment by Tarazed** [comment]
> i also want to know the loss weight of multi-task of 0.68mAP, can you share it? thanks very much! @Tarazed

just the same as what the author replied,` loss = hm_loss + wh_loss + off_loss + 0.1*angle_loss`

**Comment by 18804601171** [comment]
@Tarazed 请问用原始的centernet加一个额外的角度，除了main.py，还需要修改哪里嘛

**Comment by Tarazed** [comment]
> @Tarazed 请问用原始的centernet加一个额外的角度，除了main.py，还需要修改哪里嘛

Change the ground-truth accordingly in `base.py`
